### PR TITLE
Fix collections.abc deprecation, bump dependencies, drop py33 and pypy support

### PR DESCRIPTION
**Pull request by Kriechi**
* fix collections.abc deprecation: closes #123 
* bump dependencies for test requirements: closes #122
* add support for Python 3.7 and run tests against it
* run tests for pypy2 and pypy3 on Travis
* drop Python 3.3 support (basically impossible to test with modern CI systems and dependencies)
* adopt travis+tox+coveragerc architecture from other hyper projects

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,13 +11,13 @@ python:
 install:
   - "pip install ."
   - "pip install -r test_requirements.txt"
-  - "pip install flake8"
-before_script: "flake8 --max-complexity 10 hyperframe test"
+  - "pip install flake8 || true"
+before_script: "if [ -x "$(command -v flake8)" ] ; then flake8 --max-complexity 10 hyperframe test ; fi"
```
Comment (alexwlchan) [inline .travis.yml]: Aha, I think I see the issue. The quoting here is probably throwing the YAML parser off.

If you look at https://travis-ci.org/python-hyper/hyperframe/requests, the error it's reporting is "Could not parse .travis.yml".

Comment (Kriechi) [reply .travis.yml]: ah right - thanks! I fixed it.

**Comment by Kriechi** [comment]
@alexwlchan I stripped down the PR to the bare minimum - but I still can't get the py33 tests to pass.
Any ideas? 

Alternative solution: drop py33 entirely. Can / Should we do that in a hyperframe minor release, or do we need a 6.0 for that?

**Comment by Kriechi** [comment]
@alexwlchan would you mind reviewing again? I feel like the PR has evolved quite a bit...

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,6 @@
-.PHONY: certs publish test
-
-certs:
-	curl http://ci.kennethreitz.org/job/ca-bundle/lastSuccessfulBuild/artifact/cacerts.pem -o hyper/certs.pem
```
Comment (alexwlchan) [inline Makefile]: What was this command doing? I'm sorta reluctant to remove Make targets just for the sake of it.

File: tox.ini [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
 [testenv]
-deps= -r{toxinidir}/test_requirements.txt
-commands= py.test -n 4 --cov hyperframe {toxinidir}/test/
-
-[testenv:pypy]
-# temporarily disable coverage testing on PyPy due to performance problems
-commands= py.test -n 4 hyperframe {toxinidir}/test/
+deps = -r{toxinidir}/test_requirements.txt
+commands =
+    coverage run -m pytest --cov=hyperframe {posargs}
```
Comment (alexwlchan) [inline tox.ini]: Does this `--cov` flag get absorbed by `.coveragerc`?

File: tox.ini [hunk 4]
```diff
@@ -1,15 +1,12 @@
 [tox]
-envlist = py27, py33, py34, py35, py36, pypy, lint
```
Comment (alexwlchan) [inline tox.ini]: Why did you remove PyPy from testing?

File: Makefile (hunk 2 above)
Comment (Kriechi) [reply Makefile]: This line was here since the initial commit in this repo.
No idea what it does - but the URL does not give a valid response and is not usable.

File: tox.ini (hunk 3 above)
Comment (Kriechi) [reply tox.ini]: No - we still need to tell pytest which module it should collect coverage data on.

**Comment by KOLANICH** [comment]
>drop PyPy support
:-1:

**Comment by Kriechi** [comment]
@KOLANICH thanks for raising your voice about pypy - I didn't know if anybody would care. I have re-added pypy testing.

The Travis problems were caused by me writing invalid yaml - and are now fixed. I don't have a problem with Travis for now, so I don't see a need for switching to a different CI at the moment.

**Comment by KOLANICH** [comment]
>pypy - I didn't know if anybody would care.

In fact I use only pypy3 (I dont use python 2 at all), and use it only in some apps because currently numpy is incompatible with pypy3.

**Comment by Kriechi** [comment]
@KOLANICH ok - we never tested against pypy3 up until now... let me see if this even works...

File: tox.ini (hunk 4 above)
Comment (Kriechi) [reply tox.ini]: I have re-added pypy, and also added pypy3 testing.
